app/views/EditAry/AssessmentRegistryForm/SummaryForm/AddSummaryIssue/IssueItem/index.tsx

```diff
@@ -0,0 +1,199 @@
+import { gql, useQuery } from '@apollo/client';
+import { isNotDefined } from '@togglecorp/fujs';
+import { IoAddOutline } from 'react-icons/io5';
+
+import {
+    AssessmentRegistrySummarySubDimensionTypeEnum,
+    AssessmentRegistrySummarySubPillarTypeEnum,
+    GetSummarySubIssuesQuery,
+    GetSummarySubIssuesQueryVariables,
+    SummaryIssueSearchQuery,
+} from '#generated/types';
+
+import AddSummaryIssueModal from '../AddSummaryIssueModal';
+
+import styles from './styles.css';
+
+const GET_SUMMARY_SUB_ISSUES = gql`
+    query GetSummarySubIssues (
+        $subPillar: AssessmentRegistrySummarySubPillarTypeEnum,
+        $subDimension: AssessmentRegistrySummarySubDimensionTypeEnum,
+        $search: String,
+        $parent: ID,
+        $page: Int,
+        $pageSize: Int,
+    ) {
+        assessmentRegSummaryIssues(
+            subPillar: $subPillar,
+            subDimension: $subDimension,
+            search: $search,
+            parent: $parent,
+            pageSize: $pageSize,
+            page: $page,
+        ) {
+            page
+            results {
+                id
+                label
+                parent {
+                    id
+                    label
+                    subPillar
+                    subDimension
+                }
+                subPillar
+                subDimension
+            }
+            totalCount
+        }
+    }
+`;
+export interface Props {
+    name: string;
+    data: NonNullable<NonNullable<SummaryIssueSearchQuery['assessmentRegSummaryIssues']>['results']>[number];
+    subPillar?: AssessmentRegistrySummarySubPillarTypeEnum;
+    subDimension?: AssessmentRegistrySummarySubDimensionTypeEnum;
+}
+
+const keySelector = (d: NonNullable<NonNullable<GetSummarySubIssuesQuery['assessmentRegSummaryIssues']>['results']>[number]) => d.id;
+
+function IssueItem(props: Props) {
+    const {
+        name,
+        data,
+        subPillar,
+        subDimension,
+    } = props;
+
+    const [selected, setSelected] = useState<string | undefined>();
+    const [addIssue, setAddIssue] = useState<boolean>(false);
+
+    const variables = useMemo(
+        (): GetSummarySubIssuesQueryVariables => ({
+            subPillar: subPillar as AssessmentRegistrySummarySubPillarTypeEnum,
+            subDimension: subDimension as AssessmentRegistrySummarySubDimensionTypeEnum,
+            parent: selected,
+            page: 1,
+            pageSize: 10,
+        }), [
+            subPillar,
+            subDimension,
+            selected,
+        ],
+    );
+
+    const {
+        loading,
+        data: issuesResponse,
+        refetch,
+    } = useQuery<GetSummarySubIssuesQuery, GetSummarySubIssuesQueryVariables>(
+        GET_SUMMARY_SUB_ISSUES,
+        {
+            skip: isNotDefined(selected),
+            variables,
+        },
+    );
+
+    const handleAddNewIssue = useCallback(
+        (n: string) => {
+            setAddIssue(true);
+            setSelected(n);
```

Comment:
I don't think we need to save the id separately as we only edit one issue at a time in this component.

Reply:
The ID is saved to expand the collapsable container when clicked on the add icon